### DISCUSSION TITLE
FFmpeg should enable AudioToolbox on macOS

### DISCUSSION THREAD

**Discussion by kode54** [Homebrew/discussions#2703] [Everyday usage]
I have formulated a patch against the ffmpeg formula, which enables AudioToolbox, and also makes changes to FFmpeg source to fix AudioToolbox invocation on at least Monterey, which is apparently not as lenient when it comes to malformed output format requests. (The upstream code, even in Git master, doesn't set the bytes per frame or bytes per packet members of the ASBD structure for output format.)

I also made a second patch that makes it output appropriate lossy formats as floating point. Hoping that FFmpeg includes these eventually.

https://gist.github.com/kode54/1c9abe513473e473bb92a48f0d2431e7

This inclusion is useful at least to use the `-codec:a:0 aac_at` or such to use FFmpeg or ffplay to decode USAC files, since FFmpeg doesn't support that codec yet.

**Comment by gromgit** [comment]
Have you submitted your patches upstream? If not, please do so (https://ffmpeg.org/developer.html#Submitting-patches).

I'm not a Homebrew maintainer, but I'm sure they won't accept patches that haven't at least been reviewed and accepted by upstream. Also, nothing brings CI to a screeching halt faster than an independent patch that no longer applies to a new software version.

**Comment by DesWurstes** [comment]
@kode54 ~~Have you submitted the patches? If you don't have time I can help you.~~

Found them at https://ffmpeg.org/pipermail/ffmpeg-devel/2021-December/author.html
**Is it OK if I resend them?**

https://ffmpeg.org/pipermail/ffmpeg-devel/2022-April/thread.html#295332

May I resend the second one as well?
